### ISSUE TITLE
Labels from dev set are added to Numberer

### ISSUE THREAD

**Issue by sebpuetz** [stickeritis/sticker#28]
If there are labels in the dev set that are not in the train set, the labels get added to the numberer, leading to `NaN`s on the validation losses. Only the display of the validation batch results is affected from this, so it doesn't really have any consequences.

There's the workaround to run `sticker-prepare` on the full corpus rather than just the train set, but that adds unseen and unused classes to the softmax calculation.

Perhaps set the starting value of the numberer higher and reserve a number for unseen labels?

**Comment by danieldk** [comment]
A special value would indeed solve this. dpar has the distinction between mutable and frozen lookups, a frozen lookup can/could assign this value for any known value. I think the special value could even be the label for padding (0), since we carry around the actual sequence lengths, so we do not have to rely on this special label.

The `NaN` loss hasn't annoyed me enough yet to actually do something about this ;). Maybe after the other major changes are done.

**Comment by sebpuetz** [comment]
Once you're aware of what's going on, it's not a problem anymore. On the other hand, spending time to figure out why the loss shows up as NaN is somewhat annoying.

**Comment by danieldk** [comment]
That's true. As a temporary workaround, we could simply elide the loss for validation.
